sql/sql_update.cc

```diff
@@ -1259,6 +1307,22 @@ bool mysql_prepare_update(THD *thd, TABLE_LIST *table_list,
 
   thd->lex->allow_sum_func= 0;
 
+  if (table_list->period_conditions.name)
```

Comment:
Guess you mean good old `SYSTEM_TIME_FROM_TO`... I left the enum untouched

Reply:
But it's not `SYSTEM_TIME_FROM_TO`. We better don't mix semantics into same values.

Reply:
Not sure if there's a big difference in semantics. Does 'SELECT ... FOR SYSTEM_TIME FROM a TO b` cut row_start and row_end to range (a, b)?
e.g. for every record, `row_start = MAX(row_start, a)` and same for row_end?

Reply:
The semantics of `SYSTEM_TIME` or business time apparently different. You are currently using string comparison to differ one from the other. You can:

1. either create `PORTION_FROM_TO` enum value and check for it to discover business time.
2. create **bool** flag and rename enum prefix `SYSTEM_TIME_*` to something like `PERIOD_*`.

Reply:
Ok, now I see your point.
Unfortunately, 1 wouldn't work, because we are also going to implement [`AS OF` for application time.](https://jira.mariadb.org/browse/MDEV-16977)

Actually we shouldn't distinguish it somewhere inside. I just think that one of later features could be `UPDATE/DELETE FOR PORTION OF SYSTEM_TIME`, which is currently explicitly blocked -- and this check is the only place, where I distinguish them

Reply:
Renaming is also correct. I would like this change to be done in (SELECT...AS OF for application periods)(https://jira.mariadb.org/browse/MDEV-16977)